src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs

```diff
@@ -118,31 +118,15 @@ public override void OnStopActivity(Activity activity, object payload)
             {
                 var statusCode = this.httpStatusFetcher.Fetch(payload);
 
-                if (!statusCode.HasValue)
-                {
-                    activity.SetStatus(Status.Unknown);
-                }
-                else if (statusCode >= 200 && statusCode < 300)
-                {
-                    activity.SetStatus(Status.Ok);
-                }
-                else if (statusCode == 404)
-                {
-                    activity.SetStatus(Status.NotFound);
-                }
-                else if (statusCode == 401)
-                {
-                    activity.SetStatus(Status.Unauthenticated);
-                }
-                else if (statusCode == 400)
-                {
-                    activity.SetStatus(Status.AlreadyExists);
-                }
-                else
+                var status = Status.Error;
```

Comment:
I copied this approach from `HttpClient` instrumentation. But I think, I should set `Unset` status, and change it only if an error was occured

Reply:
@eddynaka Can you suggest here?

Reply:
We did the same thing:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/378ba54b77690f670c1ab526281d0349e1001348/src/OpenTelemetry.Api/Trace/SpanHelper.cs#L30-L40